src/pages/AuthPage.tsx

```diff
@@ -3,10 +3,20 @@ import { toast } from 'react-toastify';
 import { Card, Input, Button, Typography } from '@material-tailwind/react';
 import { authService } from '@services';
 import { useUserStore } from '@states';
+import { useState } from 'react';
+import { EyeIcon, EyeSlashIcon } from '@heroicons/react/24/outline';
+import { validateSchema } from '@utils';
+import { yupResolver } from '@hookform/resolvers/yup';
 
 export function AuthPage() {
-  const { register, handleSubmit } = useForm<LoginFormData>();
-
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<LoginFormData>({
+    resolver: yupResolver(validateSchema)
+  });
+  const [showPassword, setShowPassword] = useState<string>('password');
   const { getUserData } = useUserStore();
```

Comment:
I prefer using try catch and await instead, taking a look [here](https://www.smashingmagazine.com/2020/11/comparison-async-await-versus-then-catch/)